lib/mayaUsd/render/vp2RenderDelegate/material.cpp

```diff
@@ -353,6 +356,20 @@ struct _MaterialXData
             static const std::string env = TfGetenv("USDMTLX_PRIMARY_UV_NAME");
             _mainUvSetName = env.empty() ? UsdUtilsGetPrimaryUVSetName().GetString() : env;
 
+#if MX_COMBINED_VERSION >= 13808
+            _lobePruner = MaterialXMaya::ShaderGenUtil::LobePruner::create();
+            _lobePruner->setLibrary(_mtlxLibrary);
+
+            // TODO: Optimize published shaders.
+            // SCENARIO: User publishes a shader with a NodeGraph implementation that encapsulates a
+            // slow surface shader like OpenPBR or Standard surface. Notices the performance of the
+            // published shader is poor compared to the unpublished version. FIX: Run the LobePruner
+            // on all custom shader graphs in the _mtlxLibrary to replace the slow surfaces with
+            // optimized nodes CAVEAT: If the user has promoted all the weight attributes to the
+            // NodeGraph boundary, then no optimization will be found. This would require a change
+            // in the LobePruner to detect transitive weights. Doable, but complex. We will wait
+            // until there is sufficient demand.
```

Comment:
[Note] This could become important for the Arnold viewport implementations. The Arnold standard surface is implemented as a node graph that wraps the MaterailX standard surface and has all inputs promoted. I assume there is/will be a similar viewport implementation for the Arnold OpenPBR node.

Reply:
Good point! Currently the Arnold NodeDef doesn't inherit from the MaterialX one. It's simply:
```XML
<nodedef name="ARNOLD_ND_standard_surface" node="standard_surface" nodegroup="Shading">
```
Perhaps this can be improved on the Arnold side.